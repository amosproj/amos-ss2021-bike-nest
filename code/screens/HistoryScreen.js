import React, { useState } from 'react';
import { ImageBackground, Pressable, StyleSheet, Text, View, Image, TouchableOpacity  } from 'react-native';
import { Dimensions } from "react-native";
import Avatar from '../assets/Avatar.png'; 
import bike from '../assets/bike.png'; 
import { SimpleLineIcons } from '@expo/vector-icons';
import { AntDesign } from '@expo/vector-icons';
import global from '../components/GlobalVars';
import { mainStyles } from "../styles/MainStyles";
import BikeNest_NavigationFooter from '../components/BikeNest_NavigationFooter';


var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height


export default function HistoryScreen({ navigation }) {
  // const [myListData, setData] = useState("");

  let tryGETBooking = () => {
    console.log('start pulling reservation info');

    return fetch(global.globalIPAddress + "/booking/all", {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
    })
        .then((response) => response.json())
        .then((json) => {
          alert(JSON.stringify(json));
          console.log(json);
        })
        .catch((error) => {
            console.error(error);
        });
}

  return (
    <View style={mainStyles.container}>
      <View style={styles.historyContainer}>
        <View style={styles.containerRow}>
          <Image source={Avatar} style={styles.avatar} />
          <Text style={styles.name} >
            Max Muster </Text>
        </View>
        <TouchableOpacity onPress={() => navigation.navigate("Lock") }
          style={styles.heightBike, {
            backgroundColor: '#FFF',
            height: 230,
            width: 360,
            margin: 10,
            position: 'relative'
          }}>
          <ImageBackground
            source={bike}
            style={{
              height: 230,
              width: 360,
              position: 'absolute',
            }}
          />
          <View style={{ flex: 1, alignItems: 'flex-start', justifyContent: 'flex-start', padding: 30}}>
            <Text style={{fontSize: 16}}>Vielen Dank für die Reservierung! </Text>
            <Text style={{fontSize: 16}}>Begib dich zu folgendem BIKE NEST: {"\n"}</Text>
            <Text style={{fontWeight: 'bold', fontSize: 16}}>BIKE NEST {"\n"}Nürnberg HBF </Text>
          </View>
        </TouchableOpacity>

      <TouchableOpacity onPress={() => navigation.navigate("EditPersonalInformation")} style={styles.buttonLock}>
        <Text style={styles.buttonLockOwner}> Max Muster's bike </Text>
        <Text style={styles.buttonLockText}> locked </Text>
      </TouchableOpacity>

      <View style={styles.containerRow}>
      <TouchableOpacity style={styles.time}>
        <Text style={styles.timeText}> Zeit verging </Text>
        <SimpleLineIcons name="clock" size={24} color="black" />
        <Text style={styles.timeRecord}> 1 Tag </Text> 
      </TouchableOpacity>

      <TouchableOpacity style={styles.cost}>
        <Text style={styles.costText}> Parkkosten </Text>
        <AntDesign name="creditcard" size={24} color="black" />
        <Text style={styles.costRecord}> 50 $ </Text> 
      </TouchableOpacity>
      </View>

      <TouchableOpacity onPress={() => tryGETBooking(this)} style={styles.buttonHistory}>
        <Text style={styles.buttonHistoryText}> Frühere Reservierungen und Zahlungen </Text>
      </TouchableOpacity>
      </View>
      <BikeNest_NavigationFooter></BikeNest_NavigationFooter>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // backgroundColor: 'red',
    alignItems: 'center',
    justifyContent: 'center',
  },
  historyContainer: {
    flex: 1,
    padding: 15,
    marginTop: 20,
    margin: 10,
    justifyContent: 'space-evenly',
    position: 'relative',
  },
  containerRow:{
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    margin: 10,
  },
  avatar: {
    flex: 1,
    maxWidth: 60, 
    resizeMode: 'contain',
    marginLeft: 10,
    marginRight: 10,
  },
  name: {
    flex: 1,
    color: '#000000',
    fontSize: 18,
    fontWeight: 'bold',
    margin: 10,
  },
  welcome: {
    flex: 1,
    color: '#000000',
    fontSize: 15,
    fontWeight: 'normal',
  },
  heightBike: {
    height: 200,
    position: 'relative',           
  },
  textBike: {
    position: 'absolute',
    color: '#000000',
    fontSize:16,
    fontWeight:'bold',
    flex: 1,                    
    // backgroundColor: 'yellow',
  },
  place: {
    flex: 1,
    color: '#FFFFFF',
    fontSize:16,
    fontWeight:'bold',
  },
  nest: {
    flex: 1,
    color: '#FFFFFF',
    fontSize:14,
    fontWeight:'normal',
  },
  buttonLock: {
    flex: 1,
    backgroundColor: '#FFA500',
    borderRadius: 30,
  },
  buttonLockOwner: {
    flex: 1,
    fontSize: 20,
    color: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  }, 
  buttonLockText: {
    flex: 1,
    fontSize: 18,
    color: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  }, 
  time:{
    flex: 1,
    borderColor: '#E6E5F2',
    borderWidth:1,
    borderRadius: 30,
    padding: 10,
    maxWidth: 160,
  }, 
  timeText: {
    flex: 1,
    fontSize: 16,
    color: '#55418E',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'bold',
  }, 
  timeRecord:{
    flex: 1,
    fontSize: 16,
    color: '#FFA500',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'normal',
  }, 

  cost:{
    flex: 1,
    borderColor: '#E6E5F2',
    borderWidth:1,
    borderRadius: 30,
    padding: 10,
    maxWidth: 160,
  }, 
  costText: {
    flex: 1,
    fontSize: 16,
    color: '#55418E',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'bold',
  }, 
  costRecord:{
    flex: 1,
    fontSize: 16,
    color: '#FFA500',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'normal',
  }, 
  buttonHistory: {
    flex: 1,
    borderRadius: 30,
    borderColor: '#E6E5F2',
    borderWidth:1,
    maxHeight: 50,
  },
  buttonHistoryText: {
    flex: 1,
    fontSize: 15,
    color: '#55418E',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  }, 

});