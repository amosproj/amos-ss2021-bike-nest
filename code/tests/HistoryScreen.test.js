import React from 'react';
import renderer from 'react-test-renderer';
import HistoryScreen from '../screens/HistoryScreen';
import { NavigationContext } from "@react-navigation/native"

// fake NavigationContext value data
const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn())
}

it('renders correctly', () => {
  const tree = renderer
    .create(<NavigationContext.Provider value={navContext}>
      <HistoryScreen />
    </NavigationContext.Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});