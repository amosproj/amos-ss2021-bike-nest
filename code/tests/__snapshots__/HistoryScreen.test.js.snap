// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
        "margin": 10,
        "marginTop": 10,
        "padding": 15,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "marginLeft": 10,
              "marginRight": 10,
              "maxWidth": 60,
              "resizeMode": "contain",
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        Max Muster 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFF",
          "height": 230,
          "margin": 10,
          "opacity": 1,
          "position": "relative",
          "width": 370,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 230,
            "position": "absolute",
            "width": 370,
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 230,
                "width": 370,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "flex-start",
            "padding": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Danke für dein Vertrauen! 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Dein Fahrrad befindet sich hier: 
          

        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontStyle": "italic",
              "textDecorationLine": "underline",
            }
          }
        >
           Zeig es auf der Karte 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#E6E5F2",
            "borderRadius": 30,
            "borderWidth": 1,
            "flex": 1,
            "maxWidth": 160,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#55418E",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "bold",
              "justifyContent": "center",
            }
          }
        >
           Zeit verging 
        </Text>
        <Text />
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#FFA500",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "normal",
              "justifyContent": "center",
            }
          }
        >
           1 Tag 
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#E6E5F2",
            "borderRadius": 30,
            "borderWidth": 1,
            "flex": 1,
            "maxWidth": 160,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#55418E",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "bold",
              "justifyContent": "center",
            }
          }
        >
           Parkkosten 
        </Text>
        <Text />
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#FFA500",
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "normal",
              "justifyContent": "center",
            }
          }
        >
           50 $ 
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F28D35",
          "borderRadius": 38,
          "elevation": 3,
          "height": 55,
          "justifyContent": "center",
          "margin": 9,
          "opacity": 1,
          "padding": 18,
          "width": 328,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F2F2F2",
            "fontSize": 14,
            "height": "100%",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
         Max Muster's bike 
      </Text>
      <Text
        style={
          Object {
            "color": "#F2F2F2",
            "fontSize": 14,
            "height": "100%",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
         locked on spot X 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#E6E5F2",
          "borderRadius": 30,
          "borderWidth": 1,
          "flex": 1,
          "margin": 10,
          "maxHeight": 50,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#55418E",
            "flex": 1,
            "fontSize": 15,
            "justifyContent": "center",
            "margin": 10,
          }
        }
      >
         Frühere Reservierungen und Zahlungen 
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Profil
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          (un)lock
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Navigation
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Verlauf
        </Text>
      </View>
    </View>
  </View>
</View>
`;
